File: frontend/src/types/shop.ts
export interface Shop {
  id: number;
  platform: string;
  name: string;
  is_active: boolean;
  created_at: string;
}

export interface ShopCreateInput {
  platform: string;
  name: string;
}

export interface ShopUpdateInput {
  name?: string;
  is_active?: boolean;
  platform?: string;
}

File: backend/app/schemas/shop.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ShopRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    platform: str
    name: str
    is_active: bool
    created_at: datetime


class ShopCreate(BaseModel):
    platform: str
    name: str


class ShopUpdate(BaseModel):
    name: str | None = None
    is_active: bool | None = None
    # 現状はbase→manual(手動管理化)のみ許可。それ以外の遷移はShopServiceで拒否する
    platform: str | None = None
